### PR TITLE
[azeventhubs] readme and more tests

### PR DESCRIPTION
**Pull request by richardpark-msft**

### CODE REVIEW

File: sdk/messaging/azeventhubs/event_data_batch.go [hunk 1]
```diff
@@ -166,3 +169,35 @@ func calcActualSizeForPayload(payload []byte) uint64 {
 
 	return uint64(vbin32Overhead + len(payload))
 }
+
+func newEventDataBatch(sender amqpwrap.AMQPSenderCloser, options *NewEventDataBatchOptions) (*EventDataBatch, error) {
+	if options == nil {
+		options = &NewEventDataBatchOptions{}
+	}
+
+	if options != nil && options.PartitionID != nil && options.PartitionKey != nil {
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/event_data_batch.go:178]: The leading `options != nil` can be removed.

File: sdk/messaging/azeventhubs/producer_client.go [hunk 2]
```diff
@@ -109,43 +109,30 @@ type NewEventDataBatchOptions struct {
...
+		partitionID = *options.PartitionID
 	}
 
-	err := pc.links.Retry(ctx, exported.EventProducer, "NewEventDataBatch", getPartitionID(batch.partitionID), pc.retryOptions, func(ctx context.Context, lwid internal.LinkWithID[amqpwrap.AMQPSenderCloser]) error {
-		if options.MaxBytes == 0 {
-			batch.maxBytes = lwid.Link.MaxMessageSize()
-			return nil
-		}
+	err := pc.links.Retry(ctx, exported.EventProducer, "NewEventDataBatch", partitionID, pc.retryOptions, func(ctx context.Context, lwid internal.LinkWithID[amqpwrap.AMQPSenderCloser]) error {
+		tmpBatch, err := newEventDataBatch(lwid.Link, options)
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/producer_client.go:121]: Why is this in a retry?  It doesn't appear to perform any IO.

Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/producer_client.go:121]: It doesn't, but it requires the link to be created so we can get the max message size.

File: sdk/messaging/azeventhubs/event_data_batch.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/event_data_batch.go]: Done.

**Comment by richardpark-msft** [comment]
Just repushed, I had included some scratch changes I'd made in the azservicebus readme. Also, fixed an en-us link in the readme.md.

File: sdk/messaging/azeventhubs/README.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
+# Azure Event Hubs Client Module for Go
+
+[Azure Event Hubs](https://azure.microsoft.com/services/event-hubs/) is a big data streaming platform and event ingestion service from Microsoft. For more information about Event Hubs see this link: [link](https://docs.microsoft.com/azure/event-hubs/event-hubs-about).
+
+Use the client library `github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs` in your application to:
+
+- Send messages to an event hub.
+- Consume messages from an event hub.
+
+**NOTE**: This library is currently a beta. There may be breaking interface changes until it reaches semantic version `v1.0.0`.
```
Comment (RickWinter) [inline sdk/messaging/azeventhubs/README.md:10]:
```suggestion
**NOTE**: This library is currently a beta. There may be breaking changes until it reaches semantic version `v1.0.0`.
```

File: sdk/messaging/azeventhubs/README.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+```bash
+go get github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs
+```
+
+### Prerequisites
+
+- Go, version 1.18 or higher
+- An [Azure subscription](https://azure.microsoft.com/free/)
+- A [Event Hub namespace](https://docs.microsoft.com/azure/event-hubs/).
```
Comment (RickWinter) [inline sdk/messaging/azeventhubs/README.md:32]:
```suggestion
- An [Event Hub namespace](https://docs.microsoft.com/azure/event-hubs/).
```

File: sdk/messaging/azeventhubs/README.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+- An Event Hub. You can create an event hub in your Event Hubs Namespace using the [Azure Portal](https://docs.microsoft.com/azure/event-hubs/event-hubs-create), or the [Azure CLI](https://docs.microsoft.com/azure/event-hubs/event-hubs-quickstart-cli).
+
+### Authenticate the client
+
+The Event Hubs clients can be created using an Event Hubs connection string or a credential from the [Azure Identity package][azure_identity_pkg], like [DefaultAzureCredential][default_azure_credential].
```
Comment (RickWinter) [inline sdk/messaging/azeventhubs/README.md:37]: We want them using a credential to connect over using connection strings.  I'd move the credential to be first and move connection strings to be second in the list or perhaps as an alternative note.

```suggestion
Event Hub clients are created using an Event Hub connection string or a credential from the [Azure Identity package][azure_identity_pkg], like [DefaultAzureCredential][default_azure_credential].
```

File: sdk/messaging/azeventhubs/README.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+<!-- LINKS -->
+[azure_sdk_for_go_contributing]: https://github.com/Azure/azure-sdk-for-go/blob/main/CONTRIBUTING.md
+[azure_sdk_for_go_contributing_developer_guide]: https://github.com/Azure/azure-sdk-for-go/blob/main/CONTRIBUTING.md#developer-guide
+[azure_sdk_for_go_contributing_pull_requests]: https://github.com/Azure/azure-sdk-for-go/blob/main/CONTRIBUTING.md#pull-requests
+
+[azure_identity_pkg]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity
+[default_azure_credential]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#NewDefaultAzureCredential
+[source]: https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/messaging/azeventhubs
+[godoc]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs
+[godoc_examples]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#pkg-examples
```
Comment (chlowell) [inline sdk/messaging/azeventhubs/README.md:130]: Are we still tracking impressions?
```suggestion
[godoc_examples]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#pkg-examples

![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-go%2Fsdk%2Fmessaging%2Fazeventhubs%2FREADME.png)
```

File: sdk/messaging/azeventhubs/README.md (hunk 5 above)
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/README.md]: Done. I basically just added the connection string on as a second sentence.

File: sdk/messaging/azeventhubs/README.md (hunk 6 above)
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/README.md:131]: I think we are, good catch.

**Review by jhendrixMSFT — APPROVED** [review]

**Review by chlowell — APPROVED** [review]

**Pull request merged**